nmrfx-structure/src/main/java/org/nmrfx/structure/chemistry/miner/NanoMol.java

```diff
@@ -739,435 +665,6 @@ byte[] getBytes(byte[] inputBytes, int start) {
-        int j = 0;
-        int eVal = 0;
-        ArrayList[] eLists = new ArrayList[100];
-
-        for (int i = 0; i < nonCIndexes.length; i++) {
-            int element = nonCElements[i];
-
-            if (eLists[element] == null) {
-                eLists[element] = new ArrayList();
-            }
-
-            int index = (byte) nonCIndexes[i];
-            Integer indexInt = new Integer(index);
-            eLists[element].add(indexInt);
-        }
-
-        for (int element = 1; element < eLists.length; element++) {
-            ArrayList eList = eLists[element];
-
-            if ((eList == null) || (eList.size() == 0)) {
-                continue;
-            }
-
-            switch (element) {
-                case 1:
-                    eVal = 1;
-
-                    break;
-
-                case 7:
-                    eVal = 2;
-
-                    break;
-
-                case 8:
-                    eVal = 3;
-
-                    break;
-
-                case 9:
-                    eVal = 4;
-
-                    break;
-
-                case 15:
-                    eVal = 5;
-
-                    break;
-
-                case 16:
-                    eVal = 6;
-
-                    break;
-
-                case 17:
-                    eVal = 7;
-
-                    break;
-
-                default:
-                    eVal = 0;
-            }
-
-            if (eVal == 0) {
-                atomTypeBits.clear(j++);
-
-                for (int k = 0; k < 9; k++) {
-                    atomTypeBits.set(j++, (((element >> k) & 1) == 1));
-                }
-            } else {
-                atomTypeBits.set(j++);
-
-                for (int k = 0; k < 3; k++) {
-                    atomTypeBits.set(j++, (((eVal >> k) & 1) == 1));
-                }
-            }
-
-            for (int i = 0; i < eList.size(); i++) {
-                int index = ((Integer) eList.get(i)).intValue();
-                int nBits = 0;
-
-                if (index > 63) {
-                    nBits = 15;
-                    atomTypeBits.set(j++);
-                    atomTypeBits.set(j++);
-                } else {
-                    nBits = 7;
-                    atomTypeBits.set(j++);
-                    atomTypeBits.clear(j++);
-                }
-
-                for (int k = 0; k < nBits; k++) {
-                    atomTypeBits.set(j++, (((index >> k) & 1) == 1));
-                }
-            }
-
-            atomTypeBits.clear(j++);
-        }
-
-        atomTypeBytes = fromBitSetToBytes(atomTypeBits, j);
-    }
-
-    public void unCompressBondProps() throws Exception {
-        int j = 0;
-        BitSet bondBits = fromBytesToBitSet(bondBytes);
-        int nBondBits = getNBits(bondBytes);
-
-        for (int i = 0; i < nBondBits; i++) {
-            if (!bondBits.get(i)) {
-                nanoBonds[j++].bondProp = 1;
-            } else if (!bondBits.get(++i) && !bondBits.get(++i)) {
```

Comment:
-	Caught by sonar but I think these lines are technically not identical because the index has increased so they are retrieving a different bit, I think this function is a bit sketchy as well because of the way its incrementing i within the loop so there could be indexing errors, since this function is only called by one function, uncompressNanoRep which is not called at all, therefore I decided to remove this function and all other functions related to uncompress and compressing NanoRep, I put this in a separate commit in case removing all these methods is not wanted. I actually think the entire file could be removed as it does not seem to be used. The only other reference to NanoMol looks like it is in TurboMol.java (https://github.com/nanalysis/nmrfx/blob/7d9b8e15b120fd9ddb133501a913dcd9801e67fe/nmrfx-structure/src/main/java/org/nmrfx/structure/chemistry/miner/TurboMol.java#L293) and those methods do not appear to be implemented (https://github.com/nanalysis/nmrfx/blob/7d9b8e15b120fd9ddb133501a913dcd9801e67fe/nmrfx-structure/src/main/java/org/nmrfx/structure/chemistry/miner/TurboMol.java#L196) …Would it be okay to remove the whole file?

Reply:
I think we should actually leave NanoMol for now, and not even delete full unused methods.  We currently embed NMR information on a library of chemical compounds in NMRFx.  It would be good to have for each compound a drawing of the structure and the NanoMol code was designed to store that information in a highly compressed format.  So there is a reasonable chance that I will get around to using this code in the next few months.  If we decide to not do that, or do it in a different way, then we should delete it.

Reply:
I think we should actually leave NanoMol for now, and not even delete full unused methods.  We currently embed NMR information on a library of chemical compounds in NMRFx.  It would be good to have for each compound a drawing of the structure and the NanoMol code was designed to store that information in a highly compressed format.  So there is a reasonable chance that I will get around to using this code in the next few months.  If we decide to not do that, or do it in a different way, then we should delete it.

Reply:
@vfiack If I leave this function in, should I just add a note that the sonar identical expressions error is not actually identical, and consider it 'fixed'?

Reply:
You can mark it as false positive on sonar directly, this will update it on github as well.